### PR TITLE
feat(api): add admin panel

### PR DESCRIPTION
**Pull request by varisha-025**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (Samy-33) [inline .vscode/settings.json:1]: please add this file to .gitignore. This is supposed to be developer specific.

File: api/api/settings.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
-
+    'django.contrib.admin',
```
Comment (Samy-33) [inline api/api/settings.py:40]: move this to the top of the list for the sake of ordering.

File: api/api/settings.py [hunk 3]
```diff
@@ -95,16 +95,20 @@
 
 AUTH_PASSWORD_VALIDATORS = [
     {
-        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
```
Comment (Samy-33) [inline api/api/settings.py:98]: revert these formatting changes.

File: api/api/urls.py [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
 from django.urls.conf import include
+from django.contrib import admin
 from .views import ping_handler
+import os
 
 urlpatterns = [
     path('ping/', ping_handler, name='ping'),
-    path('api-auth/', include('rest_framework.urls', namespace='rest_framework'))
+    path('api-auth/', include(
+        'rest_framework.urls', namespace='rest_framework'))
```
Comment (Samy-33) [inline api/api/urls.py:10]: revert the formatting change.

**Review by Samy-33 — CHANGES_REQUESTED** [review]
@varisha-025 I would add a test to verify that admin only works in dev env.

**Review by Aksh-Bansal-dev — CHANGES_REQUESTED** [review]
The commit message should be in the imperative present tense.
So, it should be something like `feat(api): add admin panel`.

**Comment by varisha-025** [comment]
> @varisha-025 I would add a test to verify that admin only works in dev env.

I have done the changes as requested

**Review by Aksh-Bansal-dev — DISMISSED** [review]
LGTM.

File: .vscode/settings.json (hunk 1 above)
Comment (Samy-33) [reply .vscode/settings.json:1]: I still see this file in this PR. Use `git rm` to remove it.

**Review by Samy-33 — CHANGES_REQUESTED** [review]
I don't think the test has been added.

File: api/api/settings.py [hunk 5]
```diff
@@ -48,7 +48,7 @@
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
-    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware'
```
Comment (garg3133) [inline api/api/settings.py:51]: Restore the `,` at the end of the line. This is the preferred style of writing in Django/Python so that we can just append the elements at the end of the list without worrying if we have to put a `,` at the end of the last element.

Ref: https://stackoverflow.com/a/11597911/9890886

File: api/api/settings.py [hunk 6]
```diff
@@ -32,12 +32,12 @@
 # Application definition
 
 INSTALLED_APPS = [
+    'django.contrib.admin',
     'django.contrib.auth',
     'django.contrib.contenttypes',
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
-
```
Comment (garg3133) [inline api/api/settings.py:40]: Keep the blank line here, to create a separation between django apps and custom/3rd party apps.

**Review by garg3133 — CHANGES_REQUESTED** [review]

File: api/api/urls.py [hunk 7]
```diff
@@ -1,8 +1,13 @@
 from django.urls import path
 from django.urls.conf import include
+from django.contrib import admin
 from .views import ping_handler
+import os
```
[the remark this answers is not in the document]
Comment (Samy-33) [reply api/api/urls.py:5]: @garg3133, can you please update flake8 job in ci to check for these guidelines?

File: api/api/tests.py [hunk 8]
```diff
@@ -1,5 +1,55 @@
...
+    def testAdminEndpoint_notInDevEnv_returns404(self):
+        """Tests the /admin/ endpoint to return status code 404 when env is not dev"""
+        url = '/admin/'
+        response = self.client.get(url, follow=True)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    @pytest.mark.order(3)
+    @reload_root_urls
+    def test_dummy_toResetRootUrls(self):
+        """Should always run at the end"""
```
Comment (Samy-33) [inline api/api/tests.py:51]: add the reason behind why it needs to run at the end. -> To reset the urlconf for default environment

File: api/api/tests.py [hunk 9]
```diff
@@ -1,5 +1,55 @@
...
+        else:
+            import_module(settings.ROOT_URLCONF)
+
+        return wrapped(*args, **kwargs)
+
+    return wrapper
+
+
+class AdminTest(APITestCase):
+    """By default, pytest doesn't load .env file"""
```
Comment (Samy-33) [inline api/api/tests.py:29]: 1. Change the name of the test to `AdminEndpointTest`
2. This comment is unrelated to the the test class. The documentation should contain what this particular class is about (what it tests exactly)

File: api/api/tests.py [hunk 10]
```diff
@@ -1,5 +1,55 @@
...
+from unittest import mock
+
+
+def reload_root_urls(wrapped):
+
+    @functools.wraps(wrapped)
+    def wrapper(*args, **kwargs):
+        clear_url_caches()
+        if settings.ROOT_URLCONF in sys.modules:
+            reload(sys.modules['api.urls'])
```
Comment (Samy-33) [inline api/api/tests.py:19]: use `settings.ROOT_URLCONF` instead of `'api.urls'`

**Review by Samy-33 — CHANGES_REQUESTED** [review]
Looks good overall, just a few minor comments.

**Comment by varisha-025** [comment]
Please review @Aksh-Bansal-dev @garg3133

**Comment by garg3133** [comment]
If you can just do the following once, that'd be great:
```sh
pip install isort
isort api
```

Otherwise, looks good to me.

**Review by Aksh-Bansal-dev — APPROVED** [review]
LGTM.

**Comment by Samy-33** [comment]
@garg3133, please approve this. I have created an issue: #61 which I will take up after this is merged. As @varisha-025 is out sick, it might take long for her to push fixes.

**Comment by garg3133** [comment]
Ohh, alright!

**Review by garg3133 — APPROVED** [review]
